pyi.py

```diff
@@ -535,15 +535,14 @@ def _has_bad_hardcoded_returns(
     return_obj_name = _get_collections_abc_obj_id(returns)
     bases = {_get_collections_abc_obj_id(base_node) for base_node in classdef.bases}
 
-    return (
-        method_name == "__iter__"
-        and return_obj_name in {"Iterable", "Iterator"}
-        and "Iterator" in bases
-    ) or (
-        method_name == "__aiter__"
-        and return_obj_name in {"AsyncIterable", "AsyncIterator"}
-        and "AsyncIterator" in bases
-    )
+    if method_name == "__iter__":
+        return return_obj_name in {"Iterable", "Iterator"} and "Iterator" in bases
+    if method_name == "__aiter__":
```

Comment:
```suggestion
    elif method_name == "__aiter__":
```

Reply:
Comes to the same thing since it's straight after a `return` statement, but sure :)

Reply:
Oh oops, not sure what I was thinking

Reply:
Heh, well, it's in now, guess we're stuck with it until the next cleanup PR 😄